### PR TITLE
Fix compiler warnings and errors to enable compile on ubuntu 18.04 an…

### PR DESCRIPTION
**Pull request by mingodad**
…d several Samples can be built

### CODE REVIEW

File: debugs.c [hunk 1]
```diff
@@ -13,6 +13,7 @@ void enable_debugs ()
 	logstream = stderr;
 
 #define TEST(x)	access (DEBUGDIR #x, F_OK) == 0;
+//#define TEST(x)	1;
```
Comment (funny-falcon) [inline debugs.c:16]: excess? or really needed?

File: util.c [hunk 2]
```diff
@@ -350,15 +350,15 @@ void intremove (intnode **root, intnode *i)
 	unsigned int key = i->key;
 	intnode *n = *root;
 
-	if (!(isroot = n == i))
+	if (!((isroot = n == i))) {
```
Comment (funny-falcon) [inline util.c]: no need in another round brackets. Or they should be in other way: `if (!(isroot = (n == i))) {`

File: rexpr.c [hunk 3]
```diff
@@ -1573,7 +1574,7 @@ again:;
 		return F->t;
 
 	/* if cant get address, dont */
-	if (tmpvs [0] = R.tmpv && (!path ? !throuptr : r == 1))
+	if ((tmpvs [0] = R.tmpv && (!path ? !throuptr : r == 1)))
```
Comment (funny-falcon) [inline rexpr.c:1588]: did you check there is no bug? I mean, perhaps here should be `tmpvs[0] == R.tmv`.
I don't know really, I didn't read this code.

**Review by funny-falcon — COMMENTED** [review]
This is not my code. I just preserved it from disappearing.

It is great someone found it interesting and dig in.

I mostly approve PR. There are couple of questions.

**Comment by funny-falcon** [comment]
Note for https://github.com/funny-falcon/pyvm as well - python interpreter written in lwc. This is not mine as well, but looks like you will be interested.

If you want, I could add you as collaborator to both this repos.

File: debugs.c (hunk 1 above)
Comment (mingodad) [reply debugs.c:16]: That was needed to find the problems when compiling pyvm, probably it's better to add a command line to activate it on demand.

File: util.c (hunk 2 above)
Comment (mingodad) [reply util.c]: For sure you are right here and I did a mistake here, thanks !

File: rexpr.c (hunk 3 above)
Comment (mingodad) [reply rexpr.c:1588]: Again you seem to be right here and I did a mistake, thank you !
I have applied this fixes here https://github.com/mingodad/lwc/commit/2e07cd9597ea8d0a799197c53b573df9892964a8 and here https://github.com/mingodad/pyvm/commit/0f5117ec5132006a6047813aef7e59b4d34d5377

**Comment by mingodad** [comment]
I agree with you that it's better to join efforts than have several forks with varying degree of functionality there is also this one https://github.com/rofl0r/pyvm and you can see the discussion and several links here https://github.com/mingodad/cfront-3/issues/6

**Comment by funny-falcon** [comment]
I never gave time to this project. I'm glad I was a safe to preserve it.

Feel free to claim your repository as main repo if you'll invest time in it.

And... I'm from Russia. It is possible GitHub will close access to us in near days.

File: rexpr.c [hunk 4]
```diff
@@ -1585,7 +1585,7 @@ again:;
 		return F->t;
 
 	/* if cant get address, dont */
-	if ((tmpvs [0] = R.tmpv && (!path ? !throuptr : r == 1)))
+	if ((tmpvs [0] = R.tmpv) && (!path ? !throuptr : r == 1))
```
Comment (funny-falcon) [inline rexpr.c]: Sorry, I duplicate comment due to first time I put it on commit itself, not on PR.

I don't know how the code works. Due to "hackish" code style of original author, this line looks like buggy in old version. And I don't get which way it should work.

But this change changes the way code works. That is why I'm asking: did you really fix some bug with this change, or just put a bracket?

Comment (mingodad) [reply rexpr.c]: I agree with you, I only followed the warnings emitted by gcc and added the bracket.
Somehow lwc now build without warnings and can build all the samples although it seems that they need be revised too, one step at a time.
Again thank you and any feedback is welcome !

**Pull request merged**